In [4]:
import pandas as pd
import numpy as np

df = pd.read_csv("../data/neighbourhood-profiles-2016-140-model.csv")

print("Shape:", df.shape)

Shape: (2383, 146)


In [5]:
print(df.columns.tolist())

['_id', 'Category', 'Topic', 'Data Source', 'Characteristic', 'City of Toronto', 'Agincourt North', 'Agincourt South-Malvern West', 'Alderwood', 'Annex', 'Banbury-Don Mills', 'Bathurst Manor', 'Bay Street Corridor', 'Bayview Village', 'Bayview Woods-Steeles', 'Bedford Park-Nortown', 'Beechborough-Greenbrook', 'Bendale', 'Birchcliffe-Cliffside', 'Black Creek', 'Blake-Jones', 'Briar Hill-Belgravia', 'Bridle Path-Sunnybrook-York Mills', 'Broadview North', 'Brookhaven-Amesbury', 'Cabbagetown-South St. James Town', 'Caledonia-Fairbank', 'Casa Loma', 'Centennial Scarborough', 'Church-Yonge Corridor', 'Clairlea-Birchmount', 'Clanton Park', 'Cliffcrest', 'Corso Italia-Davenport', 'Danforth', 'Danforth East York', 'Don Valley Village', 'Dorset Park', 'Dovercourt-Wallace Emerson-Junction', 'Downsview-Roding-CFB', 'Dufferin Grove', 'East End-Danforth', 'Edenbridge-Humber Valley', 'Eglinton East', 'Elms-Old Rexdale', 'Englemount-Lawrence', 'Eringate-Centennial-West Deane', 'Etobicoke West Mall', '

In [6]:
df.head(10)

,_id,Category,Topic,Data Source,Characteristic,City of Toronto,Agincourt North,Agincourt South-Malvern West,Alderwood,Annex,...,Willowdale West,Willowridge-Martingrove-Richview,Woburn,Woodbine Corridor,Woodbine-Lumsden,Wychwood,Yonge-Eglinton,Yonge-St.Clair,York University Heights,Yorkdale-Glen Park
0,1,Neighbourhood Information,Neighbourhood Information,City of Toronto,Neighbourhood Number,NaN,129,128,20,95,...,37,7,137,64,60,94,100,97,27,31
1,2,Neighbourhood Information,Neighbourhood Information,City of Toronto,TSNS2020 Designation,NaN,No Designation,No Designation,No Designation,No Designation,...,No Designation,No Designation,NIA,No Designation,No Designation,No Designation,No Designation,No Designation,NIA,Emerging Neighbourhood
2,3,Population,Population and dwellings,Census Profile 98-316-X2016001,"Population, 2016","2,731,571","29,113","23,757","12,054","30,526",...,"16,936","22,156","53,485","12,541","7,865","14,349","11,817","12,528","27,593","14,804"
3,4,Population,Population and dwellings,Census Profile 98-316-X2016001,"Population, 2011","2,615,060","30,279","21,988","11,904","29,177",...,"15,004","21,343","53,350","11,703","7,826","13,986","10,578","11,652","27,713","14,687"
4,5,Population,Population and dwellings,Census Profile 98-316-X2016001,Population Change 2011-2016,4.50%,-3.90%,8.00%,1.30%,4.60%,...,12.90%,3.80%,0.30%,7.20%,0.50%,2.60%,11.70%,7.50%,-0.40%,0.80%
5,6,Population,Population and dwellings,Census Profile 98-316-X2016001,Total private dwellings,"1,179,057","9,371","8,535","4,732","18,109",...,"8,054","8,721","19,098","5,620","3,604","6,185","6,103","7,475","11,051","5,847"
6,7,Population,Population and dwellings,Census Profile 98-316-X2016001,Private dwellings occupied by usual residents,"1,112,929","9,120","8,136","4,616","15,934",...,"7,549","8,509","18,436","5,454","3,449","5,887","5,676","7,012","10,170","5,344"
7,8,Population,Population and dwellings,Census Profile 98-316-X2016001,Population density per square kilometre,"4,334","3,929","3,034","2,435","10,863",...,"5,820","4,007","4,345","7,838","6,722","8,541","7,162","10,708","2,086","2,451"
8,9,Population,Population and dwellings,Census Profile 98-316-X2016001,Land area in square kilometres,630.2,7.41,7.83,4.95,2.81,...,2.91,5.53,12.31,1.6,1.17,1.68,1.65,1.17,13.23,6.04
9,10,Population,Age characteristics,Census Profile 98-316-X2016001,Children (0-14 years),"398,135","3,840","3,075","1,760","2,360",...,"1,785","3,555","9,625","2,325","1,165","1,860","1,800","1,210","4,045","1,960"


In [9]:
selected_characteristics = [
    # Population
    "Population, 2016",

    # Income
    "Median total income of households in 2015 ($)",
    "Median after-tax income of households in 2015 ($)",
    "Government transfers (%)",

    # Employment
    "Employment rate",
    "Employment rate (Males)",
    "Employment rate (Females)",

    # Education
    "No certificate, diploma or degree",
    "Postsecondary certificate, diploma or degree",
    "Bachelor's degree",
    "Master's degree",

    # Housing
    "Rate of core housing need",

    # Target variable
    "Prevalence of low income based on the Low-income measure, after tax (LIM-AT) (%)"]

In [10]:
for item in selected_characteristics:
    matches = df[df["Characteristic"].astype(str).str.contains(
        item,
        case=False,
        na=False)]

    print("\n", item)
    print("Matches:", len(matches))


 Population, 2016
Matches: 1

 Median total income of households in 2015 ($)
Matches: 0

 Median after-tax income of households in 2015 ($)
Matches: 0

 Government transfers (%)
Matches: 0

 Employment rate
Matches: 6

 Employment rate (Males)
Matches: 0

 Employment rate (Females)
Matches: 0

 No certificate, diploma or degree
Matches: 2

 Postsecondary certificate, diploma or degree
Matches: 8

 Bachelor's degree
Matches: 2


C:\Users\local_ksalyamova\Temp\ipykernel_13812\2785299153.py:2: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  matches = df[df["Characteristic"].astype(str).str.contains(
C:\Users\local_ksalyamova\Temp\ipykernel_13812\2785299153.py:2: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  matches = df[df["Characteristic"].astype(str).str.contains(
C:\Users\local_ksalyamova\Temp\ipykernel_13812\2785299153.py:2: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  matches = df[df["Characteristic"].astype(str).str.contains(
C:\Users\local_ksalyamova\Temp\ipykernel_13812\2785299153.py:2: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  matches = df[df["Charact


 Master's degree
Matches: 2

 Rate of core housing need
Matches: 1

 Prevalence of low income based on the Low-income measure, after tax (LIM-AT) (%)
Matches: 0


C:\Users\local_ksalyamova\Temp\ipykernel_13812\2785299153.py:2: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  matches = df[df["Characteristic"].astype(str).str.contains(


In [11]:
df[["Category","Topic","Characteristic"]].head()

,Category,Topic,Characteristic
0,Neighbourhood Information,Neighbourhood Information,Neighbourhood Number
1,Neighbourhood Information,Neighbourhood Information,TSNS2020 Designation
2,Population,Population and dwellings,"Population, 2016"
3,Population,Population and dwellings,"Population, 2011"
4,Population,Population and dwellings,Population Change 2011-2016


In [12]:
for value in [
    "Population, 2016",
    "Median total income of households in 2015 ($)",
    "Median after-tax income of households in 2015 ($)",
    "Government transfers (%)",
    "Employment rate",
    "Employment rate (Males)",
    "Employment rate (Females)",
    "No certificate, diploma or degree",
    "Postsecondary certificate, diploma or degree",
    "Bachelor's degree",
    "Master's degree",
    "Rate of core housing need",
    "Prevalence of low income based on the Low-income measure, after tax (LIM-AT) (%)"]:
    
    matches = df[df["Characteristic"] == value]

    print("\n", value)
    print("Matches:", len(matches))


 Population, 2016
Matches: 1

 Median total income of households in 2015 ($)
Matches: 0

 Median after-tax income of households in 2015 ($)
Matches: 0

 Government transfers (%)
Matches: 0

 Employment rate
Matches: 1

 Employment rate (Males)
Matches: 1

 Employment rate (Females)
Matches: 1

 No certificate, diploma or degree
Matches: 0

 Postsecondary certificate, diploma or degree
Matches: 0

 Bachelor's degree
Matches: 0

 Master's degree
Matches: 0

 Rate of core housing need
Matches: 1

 Prevalence of low income based on the Low-income measure, after tax (LIM-AT) (%)
Matches: 1


In [13]:
for col in df["Characteristic"]:
    if "Median total income" in str(col):
        print(repr(col))

'    Median total income in 2015 among recipients ($)'
'      Median total income of one-person households in 2015 ($)'
'  Median total income of households in 2015 ($)'
'      Median total income of two-or-more-person households in 2015 ($)'
'  Median total income of economic families in 2015 ($)'
'  Median total income of couple economic families without children or other relatives in 2015 ($)'
'  Median total income of couple economic families with children in 2015 ($)'
'  Median total income of lone-parent economic families in 2015 ($)'
'  Median total income in 2015 for persons aged 15 years and over not in economic families ($)'


In [14]:
df["Characteristic_Clean"] = (
    df["Characteristic"]
    .astype(str)
    .str.strip())

In [15]:
df[df["Characteristic_Clean"] == "Median total income of households in 2015 ($)"]

,_id,Category,Topic,Data Source,Characteristic,City of Toronto,Agincourt North,Agincourt South-Malvern West,Alderwood,Annex,...,Willowridge-Martingrove-Richview,Woburn,Woodbine Corridor,Woodbine-Lumsden,Wychwood,Yonge-Eglinton,Yonge-St.Clair,York University Heights,Yorkdale-Glen Park,Characteristic_Clean
1018,1019,Income,Income of households in 2015,Census Profile 98-316-X2016001,Median total income of households in 2015 ($),"65,829",NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Median total income of households in 2015 ($)


In [16]:
variables = [

    "Population, 2016",

    "Median total income of households in 2015 ($)",
    "Median after-tax income of households in 2015 ($)",
    "Government transfers (%)",

    "Employment rate",
    "Employment rate (Males)",
    "Employment rate (Females)",

    "No certificate, diploma or degree",
    "Postsecondary certificate, diploma or degree",
    "Bachelor's degree",
    "Master's degree",

    "Rate of core housing need",

    "Prevalence of low income based on the Low-income measure, after tax (LIM-AT) (%)"
]

for v in variables:
    matches = df[df["Characteristic_Clean"] == v]
    print(f"{v} -> {len(matches)}")

Population, 2016 -> 1
Median total income of households in 2015 ($) -> 1
Median after-tax income of households in 2015 ($) -> 1
Government transfers (%) -> 1
Employment rate -> 1
Employment rate (Males) -> 1
Employment rate (Females) -> 1
No certificate, diploma or degree -> 2
Postsecondary certificate, diploma or degree -> 4
Bachelor's degree -> 2
Master's degree -> 2
Rate of core housing need -> 1
Prevalence of low income based on the Low-income measure, after tax (LIM-AT) (%) -> 1


In [17]:

df["Characteristic_Clean"] = df["Characteristic"].astype(str).str.strip()

selected_rows = df[
    (
        df["Characteristic_Clean"].isin([
            "Population, 2016",
            "Median total income of households in 2015 ($)",
            "Median after-tax income of households in 2015 ($)",
            "Government transfers (%)",
            "Employment rate",
            "Employment rate (Males)",
            "Employment rate (Females)",
            "Rate of core housing need",
            "Prevalence of low income based on the Low-income measure, after tax (LIM-AT) (%)"
        ])
    )
    |
    (
        (df["Topic"] == "Highest certificate, diploma or degree")
        &
        (df["Characteristic_Clean"].isin([
            "No certificate, diploma or degree",
            "Postsecondary certificate, diploma or degree",
            "Bachelor's degree",
            "Master's degree"
        ]))
    )
]

print(selected_rows.shape)
selected_rows[["Category","Topic","Characteristic_Clean"]]

(17, 147)


,Category,Topic,Characteristic_Clean
2,Population,Population and dwellings,"Population, 2016"
976,Income,Income of individuals in 2015,Government transfers (%)
1018,Income,Income of households in 2015,Median total income of households in 2015 ($)
1019,Income,Income of households in 2015,Median after-tax income of households in 2015 ($)
1130,Income,Low income in 2015,Prevalence of low income based on the Low-inco...
1211,Education,"Highest certificate, diploma or degree",Master's degree
1675,Education,"Highest certificate, diploma or degree","Postsecondary certificate, diploma or degree"
1681,Housing,Core housing need,Rate of core housing need
1703,Education,"Highest certificate, diploma or degree","No certificate, diploma or degree"
1710,Education,"Highest certificate, diploma or degree",Bachelor's degree


In [18]:
selected_rows.loc[
    selected_rows["Topic"] == "Highest certificate, diploma or degree",
    ["Data Source", "Characteristic_Clean"]
]

,Data Source,Characteristic_Clean
1211,Census Profile 98-316-X2016001,Master's degree
1675,Census Profile 98-316-X2016001,"Postsecondary certificate, diploma or degree"
1703,Census Profile 98-316-X2016001,"No certificate, diploma or degree"
1710,Census Profile 98-316-X2016001,Bachelor's degree
1715,Census Profile 98-316-X2016001,"No certificate, diploma or degree"
1717,Census Profile 98-316-X2016001,"Postsecondary certificate, diploma or degree"
1720,Census Profile 98-316-X2016001,Master's degree
1725,Census Profile 98-316-X2016001,Bachelor's degree


In [19]:
df.loc[
    [1211,1675,1703,1710,1715,1717,1720,1725],
    ["Characteristic_Clean","City of Toronto"]
]

,Characteristic_Clean,City of Toronto
1211,Master's degree,192120
1675,"Postsecondary certificate, diploma or degree",1356360
1703,"No certificate, diploma or degree",377340
1710,Bachelor's degree,534610
1715,"No certificate, diploma or degree",159795
1717,"Postsecondary certificate, diploma or degree",1077935
1720,Master's degree,166405
1725,Bachelor's degree,433620


In [20]:
df.loc[
    [1211,1675,1703,1710,1715,1717,1720,1725],
    ["Characteristic_Clean","City of Toronto","Agincourt North"]
]

,Characteristic_Clean,City of Toronto,Agincourt North
1211,Master's degree,192120,890
1675,"Postsecondary certificate, diploma or degree",1356360,10985
1703,"No certificate, diploma or degree",377340,6550
1710,Bachelor's degree,534610,4380
1715,"No certificate, diploma or degree",159795,3335
1717,"Postsecondary certificate, diploma or degree",1077935,7870
1720,Master's degree,166405,735
1725,Bachelor's degree,433620,3090


In [21]:
df.loc[
    [1211,1675,1703,1710,1715,1717,1720,1725],
    ["Category","Topic","Characteristic","Data Source"]
]

,Category,Topic,Characteristic,Data Source
1211,Education,"Highest certificate, diploma or degree",Master's degree,Census Profile 98-316-X2016001
1675,Education,"Highest certificate, diploma or degree","Postsecondary certificate, diploma or degree",Census Profile 98-316-X2016001
1703,Education,"Highest certificate, diploma or degree","No certificate, diploma or degree",Census Profile 98-316-X2016001
1710,Education,"Highest certificate, diploma or degree",Bachelor's degree,Census Profile 98-316-X2016001
1715,Education,"Highest certificate, diploma or degree","No certificate, diploma or degree",Census Profile 98-316-X2016001
1717,Education,"Highest certificate, diploma or degree","Postsecondary certificate, diploma or degree",Census Profile 98-316-X2016001
1720,Education,"Highest certificate, diploma or degree",Master's degree,Census Profile 98-316-X2016001
1725,Education,"Highest certificate, diploma or degree",Bachelor's degree,Census Profile 98-316-X2016001


In [22]:
pd.set_option("display.max_colwidth", None)

df.loc[
    [1211,1675,1703,1710,1715,1717,1720,1725],
    ["Characteristic"]
]

,Characteristic
1211,Master's degree
1675,"Postsecondary certificate, diploma or degree"
1703,"No certificate, diploma or degree"
1710,Bachelor's degree
1715,"No certificate, diploma or degree"
1717,"Postsecondary certificate, diploma or degree"
1720,Master's degree
1725,Bachelor's degree


In [24]:
selected_rows = df[
    df["Characteristic_Clean"].isin([
        "Population, 2016",
        "Median total income of households in 2015 ($)",
        "Median after-tax income of households in 2015 ($)",
        "Government transfers (%)",
        "Employment rate",
        "Employment rate (Males)",
        "Employment rate (Females)",
        "Rate of core housing need",
        "Prevalence of low income based on the Low-income measure, after tax (LIM-AT) (%)"])]

print(selected_rows.shape)

selected_rows[["Category","Topic","Characteristic_Clean"]]

(9, 147)


,Category,Topic,Characteristic_Clean
2,Population,Population and dwellings,"Population, 2016"
976,Income,Income of individuals in 2015,Government transfers (%)
1018,Income,Income of households in 2015,Median total income of households in 2015 ($)
1019,Income,Income of households in 2015,Median after-tax income of households in 2015 ($)
1130,Income,Low income in 2015,"Prevalence of low income based on the Low-income measure, after tax (LIM-AT) (%)"
1681,Housing,Core housing need,Rate of core housing need
1889,Labour,Labour force status,Employment rate
1897,Labour,Labour force status,Employment rate (Males)
1905,Labour,Labour force status,Employment rate (Females)


In [30]:
neighbourhood_cols = df.columns[5:]

analysis_df = pd.DataFrame({"Neighbourhood": neighbourhood_cols})

analysis_df.head()

,Neighbourhood
0,City of Toronto
1,Agincourt North
2,Agincourt South-Malvern West
3,Alderwood
4,Annex


In [32]:
def get_values(characteristic):

    row = selected_rows[
        selected_rows["Characteristic_Clean"] == characteristic]

    return row.iloc[0, 5:].values

In [34]:
analysis_df["Population_2016"] = get_values("Population, 2016")

analysis_df["Median_Income"] = get_values(
    "Median total income of households in 2015 ($)")

analysis_df["Median_AfterTax_Income"] = get_values(
    "Median after-tax income of households in 2015 ($)")

analysis_df["Government_Transfers_Pct"] = get_values(
    "Government transfers (%)")

analysis_df["Employment_Rate"] = get_values(
    "Employment rate")

analysis_df["Employment_Rate_Male"] = get_values(
    "Employment rate (Males)")

analysis_df["Employment_Rate_Female"] = get_values(
    "Employment rate (Females)")

analysis_df["Core_Housing_Need_Pct"] = get_values(
    "Rate of core housing need")

analysis_df["Low_Income_Pct"] = get_values(
    "Prevalence of low income based on the Low-income measure, after tax (LIM-AT) (%)")

In [35]:
analysis_df.shape

(142, 10)

In [36]:
analysis_df.head()

,Neighbourhood,Population_2016,Median_Income,Median_AfterTax_Income,Government_Transfers_Pct,Employment_Rate,Employment_Rate_Male,Employment_Rate_Female,Core_Housing_Need_Pct,Low_Income_Pct
0,City of Toronto,"2,731,571","65,829","58,264",9.4,59.3,63.8,55.2,23.3,20.2
1,Agincourt North,"29,113",NaN,NaN,NaN,50,54.5,46,NaN,21.3
2,Agincourt South-Malvern West,"23,757",NaN,NaN,NaN,53.2,58.2,48.7,NaN,22.6
3,Alderwood,"12,054",NaN,NaN,NaN,62.4,67.1,57.9,NaN,8.2
4,Annex,"30,526",NaN,NaN,NaN,65.8,68.9,63.1,NaN,18.3


In [37]:
selected_rows[
    selected_rows["Characteristic_Clean"] ==
    "Median total income of households in 2015 ($)"
][["Category","Topic","Data Source","Characteristic_Clean"]]

,Category,Topic,Data Source,Characteristic_Clean
1018,Income,Income of households in 2015,Census Profile 98-316-X2016001,Median total income of households in 2015 ($)


In [38]:
selected_rows.loc[
    1018,
    ["City of Toronto",
     "Agincourt North",
     "Agincourt South-Malvern West",
     "Alderwood",
     "Annex"]]

City of Toronto                 65,829
Agincourt North                    NaN
Agincourt South-Malvern West       NaN
Alderwood                          NaN
Annex                              NaN
Name: 1018, dtype: object

In [39]:
selected_rows.loc[
    1681,
    ["City of Toronto",
     "Agincourt North",
     "Agincourt South-Malvern West",
     "Alderwood",
     "Annex"]]

City of Toronto                 23.3
Agincourt North                  NaN
Agincourt South-Malvern West     NaN
Alderwood                        NaN
Annex                            NaN
Name: 1681, dtype: object

In [40]:
df[
    df["Characteristic"].astype(str).str.contains(
        "Median total income of households",
        case=False,
        na=False
    )
][["Topic","Characteristic"]]

,Topic,Characteristic
1018,Income of households in 2015,Median total income of households in 2015 ($)


In [42]:
df[
    df["Characteristic"].astype(str).str.contains(
        "core housing need",
        case=False,
        na=False
    )
][["Topic","Characteristic"]]

,Topic,Characteristic
1681,Core housing need,Rate of core housing need
1682,Core housing need,Number of households in core housing need
1684,Core housing need,"Core housing need, Unsuitable only"
1685,Core housing need,"Core housing need, Inadequate only"
1686,Core housing need,"Core housing need, Unaffordable only"
1687,Core housing need,"Core housing need, two or more dimensions"


In [43]:
analysis_df.notnull().sum()

Neighbourhood               142
Population_2016             142
Median_Income                 2
Median_AfterTax_Income        2
Government_Transfers_Pct      2
Employment_Rate             142
Employment_Rate_Male        142
Employment_Rate_Female      142
Core_Housing_Need_Pct         2
Low_Income_Pct              142
dtype: int64

In [44]:
analysis_df_clean = analysis_df.drop(
    columns=[
        "Median_Income",
        "Median_AfterTax_Income",
        "Government_Transfers_Pct",
        "Core_Housing_Need_Pct"
    ])

analysis_df_clean.shape

(142, 6)

In [45]:
analysis_df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 142 entries, 0 to 141
Data columns (total 6 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Neighbourhood           142 non-null    object
 1   Population_2016         142 non-null    object
 2   Employment_Rate         142 non-null    object
 3   Employment_Rate_Male    142 non-null    object
 4   Employment_Rate_Female  142 non-null    object
 5   Low_Income_Pct          142 non-null    object
dtypes: object(6)
memory usage: 6.8+ KB


In [46]:
analysis_df_clean.describe()

,Neighbourhood,Population_2016,Employment_Rate,Employment_Rate_Male,Employment_Rate_Female,Low_Income_Pct
count,142,142,142,142,142,142
unique,142,142,108,106,114,115
top,City of Toronto,"2,731,571",55.9,70.8,63.1,18.3
freq,1,1,4,5,4,3


In [47]:
from ydata_profiling import ProfileReport

profile = ProfileReport(
    analysis_df_clean,
    title="Toronto Neighbourhood Socioeconomic EDA",
    minimal=True)

profile.to_file("../reports/Toronto_EDA_Final.html")

C:\Users\local_ksalyamova\Temp\ipykernel_13812\4294916145.py:1: DeprecationWarning: 
    `import ydata_profiling` is deprecated and will not receive more updates. 
    Please install fg-data-profiling via `pip install fg-data-profiling` and use `import data_profiling` instead.
    
  from ydata_profiling import ProfileReport


Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

100%|██████████| 6/6 [00:00<00:00, 35.29it/s]


Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [48]:
analysis_df_clean.to_csv(
    "../data/toronto_neighbourhood_analysis.csv",
    index=False)

In [49]:
analysis_df_clean.describe()

,Neighbourhood,Population_2016,Employment_Rate,Employment_Rate_Male,Employment_Rate_Female,Low_Income_Pct
count,142,142,142,142,142,142
unique,142,142,108,106,114,115
top,City of Toronto,"2,731,571",55.9,70.8,63.1,18.3
freq,1,1,4,5,4,3


In [50]:
analysis_df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 142 entries, 0 to 141
Data columns (total 6 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Neighbourhood           142 non-null    object
 1   Population_2016         142 non-null    object
 2   Employment_Rate         142 non-null    object
 3   Employment_Rate_Male    142 non-null    object
 4   Employment_Rate_Female  142 non-null    object
 5   Low_Income_Pct          142 non-null    object
dtypes: object(6)
memory usage: 6.8+ KB


In [51]:
analysis_df_clean.head()

,Neighbourhood,Population_2016,Employment_Rate,Employment_Rate_Male,Employment_Rate_Female,Low_Income_Pct
0,City of Toronto,"2,731,571",59.3,63.8,55.2,20.2
1,Agincourt North,"29,113",50,54.5,46,21.3
2,Agincourt South-Malvern West,"23,757",53.2,58.2,48.7,22.6
3,Alderwood,"12,054",62.4,67.1,57.9,8.2
4,Annex,"30,526",65.8,68.9,63.1,18.3


In [52]:
analysis_df_clean.dtypes

Neighbourhood             object
Population_2016           object
Employment_Rate           object
Employment_Rate_Male      object
Employment_Rate_Female    object
Low_Income_Pct            object
dtype: object

In [53]:
for col in analysis_df_clean.columns[1:]:
    analysis_df_clean[col] = (
        analysis_df_clean[col]
        .astype(str)
        .str.replace(",", "", regex=False))

    analysis_df_clean[col] = pd.to_numeric(
        analysis_df_clean[col],
        errors="coerce")

In [54]:
analysis_df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 142 entries, 0 to 141
Data columns (total 6 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Neighbourhood           142 non-null    object 
 1   Population_2016         141 non-null    float64
 2   Employment_Rate         141 non-null    float64
 3   Employment_Rate_Male    141 non-null    float64
 4   Employment_Rate_Female  141 non-null    float64
 5   Low_Income_Pct          141 non-null    float64
dtypes: float64(5), object(1)
memory usage: 6.8+ KB


In [56]:
from ydata_profiling import ProfileReport

profile = ProfileReport(
    analysis_df_clean,
    title="Toronto Neighbourhood Socioeconomic EDA",
    minimal=True)

profile.to_file("../reports/Toronto_EDA_Final_V2.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

100%|██████████| 6/6 [00:00<00:00, 38.22it/s]


Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

# EDA Summary

This exploratory data analysis examines socioeconomic characteristics across Toronto neighbourhoods using selected indicators from the Toronto Neighbourhood Profiles dataset.

The final analysis dataset contains 142 neighbourhoods and the following variables:

- Population (2016)
- Employment Rate
- Employment Rate (Male)
- Employment Rate (Female)
- Low Income Percentage

The dataset was cleaned, transformed into neighbourhood-level observations, converted to numeric formats, and analyzed using YData Profiling to generate an automated EDA report.

The resulting report provides descriptive statistics, distributions, missing-value analysis, and variable relationships to support further socioeconomic analysis.